### PR TITLE
Expanded contributor docs on getting feedback from the wider community.

### PR DESCRIPTION
**Pull request by mharyam**
#### Trac ticket number
N/A

Discussed on the forum: https://forum.djangoproject.com/t/feedback-needed-updating-documentation-contribution-guidelines/33448

#### Branch description
Based on the forum discussions, this improves the contribution guidelines by making them more descriptive and clarifying what the guidelines for contributing are.

Screenshot 2024-08-28 at 10 10 56 PM

#### Checklist
- [x] This PR targets the `main` branch. 
- [x] The commit message is written in past tense, mentions the ticket number, and ends with a period.
- [ ] I have checked the "Has patch" ticket flag in the Trac system.
- [ ] I have added or updated relevant tests.
- [ ] I have added or updated relevant docs, including release notes if applicable.
- [x] I have attached screenshots in both light and dark modes for any UI changes.

### CODE REVIEW

File: docs/internals/contributing/writing-code/submitting-patches.txt [hunk 1]
```diff
@@ -140,20 +140,62 @@ Regardless of the way you submit your work, follow these steps.
...
-=========================
+Contributions which require community feedback
+==============================================
 
-A "non-trivial" contribution is one that is more than a small bug fix. It's a
-change that introduces new Django functionality and makes some sort of design
-decision.
+A patch that introduces Django functionality and makes some sort of design
+decision requires a wider community discussion. Especially if the approach
+involves a deprecation or potentially introducing breaking changes.
```
Comment (sarahboyce) [inline docs/internals/contributing/writing-code/submitting-patches.txt:148]: I'm wondering if we should turn this into a bullet list and add the "needs strong consensus" examples. 
Another example beyond depredations and breaking changes would be "adding a new setting" (as many people suggest to "just" add a setting and we try to avoid having too many settings in Django)

File: docs/internals/contributing/writing-code/submitting-patches.txt [hunk 2]
```diff
@@ -140,20 +140,62 @@ Regardless of the way you submit your work, follow these steps.
...
 
-If you're not sure whether your contribution should be considered non-trivial,
-ask on the ticket for opinions.
+Django Forum or the django-developers mailing list
+--------------------------------------------------
+
+You can make a proposal on the `Django Forum`_ or |django-developers| mailing list.
+
+You should explain the need for the change, go into details of the approach
+and discuss alternatives. Please include a link to such discussions in your patch.
```
Comment (sarahboyce) [inline docs/internals/contributing/writing-code/submitting-patches.txt:160]: (thinking aloud) there is an FAQ section which has a "how/when can I remind people about a change": https://docs.djangoproject.com/en/5.1/faq/contributing/#when-and-how-might-i-remind-the-team-of-a-change-i-care-about
This isn't an exact match but some people need to know it might take time to get feedback :thinking:

File: docs/internals/contributing/writing-code/submitting-patches.txt [hunk 3]
```diff
@@ -140,20 +140,62 @@ Regardless of the way you submit your work, follow these steps.
...
+
+Once this package becomes stable and there are clear benefits of
+incorporating aspects into Django core, starting a discussion on the
+`Django Forum`_ or |django-developers| mailing list would be the next step.
+
+Django Enhancement Proposal (DEP)
+---------------------------------
+
+Similar to Python’s PEPs, Django has `Django Enhancement Proposals`_ or DEPs.
+A DEP is a design document providing information to the Django community,
```
Comment (sarahboyce) [inline docs/internals/contributing/writing-code/submitting-patches.txt:183]: Looks like the spell checker failed and we might need to add "DEP" to `docs/spelling_wordlist` (see https://docs.djangoproject.com/en/5.1/internals/contributing/writing-documentation/#spelling-check)

File: docs/internals/contributing/writing-code/submitting-patches.txt [hunk 4]
```diff
@@ -140,20 +140,62 @@ Regardless of the way you submit your work, follow these steps.
...
+
+DEPs are primary mechanisms for proposing and collecting community input on
+major new features, and for documenting design decisions that have gone into Django.
+
+Before considering writing a DEP, its recommeneded to first open a discussion
+on the `Django Forum`_ or |django-developers| mailing list.
+
+This allows the community to provide feedback and helps refine the proposal.
+Once the DEP is ready the :ref:`Steering Council <steering-council>`
+votes on whether to accept it.
```
Comment (sarahboyce) [inline docs/internals/contributing/writing-code/submitting-patches.txt:195]: I wonder if we should add links to any specific DEPs in final status as examples, such as
- [DEP 181: ORM Expressions](https://github.com/django/deps/blob/main/final/0181-orm-expressions.rst)
- [DEP 182: Multiple Template Engines](https://github.com/django/deps/blob/main/final/0182-multiple-template-engines.rst)
- [DEP 201: Simplified routing syntax](https://github.com/django/deps/blob/main/final/0201-simplified-routing-syntax.rst)

File: docs/internals/contributing/writing-code/submitting-patches.txt [hunk 5]
```diff
@@ -140,20 +140,62 @@ Regardless of the way you submit your work, follow these steps.
...
+
+Therefore, in such cases, it is better to create a third-party package which
+seeks user feedback. The public API can be iterated on and you can validate
+that there is a need for such a feature.
+
+Once this package becomes stable and there are clear benefits of
+incorporating aspects into Django core, starting a discussion on the
+`Django Forum`_ or |django-developers| mailing list would be the next step.
+
+Django Enhancement Proposal (DEP)
```
Comment (sarahboyce) [inline docs/internals/contributing/writing-code/submitting-patches.txt:179]: @django/steering-council I would love to have you add your thoughts to this PR as this would be the first time we mention DEPs in the contributing docs

**Review by sarahboyce — COMMENTED** [review]
Thank you @mharyam :star: 
I've added some comments which are more "thinking aloud" than concrete suggestions - so I'd love to hear your thoughts

File: docs/internals/contributing/writing-code/submitting-patches.txt (hunk 3 above)
Comment (mharyam) [reply docs/internals/contributing/writing-code/submitting-patches.txt:183]: Ah, Thank you. 
Will add it

File: docs/internals/contributing/writing-code/submitting-patches.txt (hunk 1 above)
Comment (mharyam) [reply docs/internals/contributing/writing-code/submitting-patches.txt:148]: Yes, Will prefer that too. It would be easier to read if we have it as a list

Comment (mharyam) [reply docs/internals/contributing/writing-code/submitting-patches.txt:148]: I have changed it to a list, and it looks better and more readable. Thank you. 

I'm not  sure I understand the examples you mentioned, could you help with how it should look like?

File: docs/internals/contributing/writing-code/submitting-patches.txt (hunk 2 above)
Comment (mharyam) [reply docs/internals/contributing/writing-code/submitting-patches.txt]: > You should explain the need for the change, go into details of the approach
> and discuss alternatives. Please include a link to such discussions in your patch.
> Keep in mind that it might take some time to receive feedback, but you can remind people if it takes too long.[Add the link to the forum here]

Does this work?

File: docs/internals/contributing/writing-code/submitting-patches.txt (hunk 5 above)
Comment (mharyam) [reply docs/internals/contributing/writing-code/submitting-patches.txt]: @sarahboyce Thank you reviewing.
I have updated changes, and this is what it looks like now 

Screenshot 2024-09-12 at 2 32 43 PM

I’m wondering if the list items are too wordy because we have some paragraphs within them, like in DEP, where we have about three. Is this okay, or should we consider making it more concise? I’m also thinking it might be better to make the list items more descriptive. What do you think?

File: docs/internals/contributing/writing-code/submitting-patches.txt [hunk 6]
```diff
@@ -140,20 +140,66 @@ Regardless of the way you submit your work, follow these steps.
...
-If you're not sure whether your contribution should be considered non-trivial,
-ask on the ticket for opinions.
-
-.. _Django Forum: https://forum.djangoproject.com/
+Contributions which require community feedback
+==============================================
+
+A patch that introduces Django functionality and makes some sort of design
+decision requires a wider community discussion. Especially if the approach
+involves a deprecation or potentially introducing breaking changes.
```
Comment (cliff688) [inline docs/internals/contributing/writing-code/submitting-patches.txt:148]: Adding settings in the list of non-trivial patches as per @sarahboyce's suggestion, maybe something like
```
A patch that introduces Django functionality and makes a design
decision requires wider community discussion. Examples of this include
adding new settings, adding new features, deprecations, and potentially
breaking changes.
```
wrapped at 79 characters

**Review by sarahboyce — APPROVED** [review]
Thank you @mharyam :star:
